src/benchmark_register.cc

```diff
@@ -402,9 +412,17 @@ Benchmark* Benchmark::UseRealTime() {
 }
 
 Benchmark* Benchmark::UseManualTime() {
+  return UseManualTime("s", TimeUnit::kNanosecond, BenchmarkTime::SecondsCost);
+}
+
+Benchmark* Benchmark::UseManualTime(
+    std::string name, TimeUnit unit_multiplier,
+    BenchmarkTimeCostFunc* cost_function) {
   CHECK(!use_real_time_)
       << "Cannot set UseRealTime and UseManualTime simultaneously.";
   use_manual_time_ = true;
+  time_.SetUnitString(unit_multiplier, name);
+  time_.to_cost_in_seconds_ = cost_function ? cost_function : BenchmarkTime::SecondsCost;
```

Comment:
you can always set this to cost_function. If it's left unset, it'll be defaulted to SecondsCost. If the user passed nullptr then they likely made a mistake and you should CHECK that.